src/App.tsx

```diff
@@ -1,61 +1,139 @@
+import React, { useState } from 'react';
 import './App.scss';
 
-// import usersFromServer from './api/users';
-// import todosFromServer from './api/todos';
+import usersFromServer from './api/users';
+import todosFromServer from './api/todos';
+
+import { TodoList } from './components/TodoList';
+
+export type Todo = {
+  id: number;
+  title: string;
+  completed: boolean;
+  userId: number;
+};
 
 export const App = () => {
+  // #region inputs
+  const [title, setTitle] = useState('');
+  const [titleError, setTitleError] = useState(false);
+
+  const [userId, setUserId] = useState(0);
+  const [userIdError, setUserIdError] = useState(false);
+
+  const handleTitleChange = (event: React.ChangeEvent<HTMLInputElement>) => {
+    const sanitizedValue = event.target.value.replace(
+      /[^a-zA-Zа-яА-ЯіІєЄїЇґҐ0-9\s]/g,
+      '',
+    );
+
+    setTitle(sanitizedValue);
+    setTitleError(false);
+  };
+
+  const handleUserIdChange = (event: React.ChangeEvent<HTMLSelectElement>) => {
+    setUserId(+event.target.value);
+    setUserIdError(false);
+  };
+  // #endregion
+
+  // #region todos
+  const [currentTodos, setCurrentTodos] = useState<Todo[]>([
+    ...todosFromServer,
+  ]);
+
+  const addTodo = (todo: Todo) => {
+    setCurrentTodos(todos => [...todos, todo]);
+  };
+
+  // #endregion
+
+  // #region return function
+  const clearInputs = () => {
+    setTitle('');
+    setUserId(0);
+    setTitleError(false);
+    setUserIdError(false);
+  };
+  // #endregion
+
+  // #region create id function
+  const uniqueId = () => {
+    const maxId = Math.max(...currentTodos.map(todo => todo.id));
+
+    return maxId + 1;
+  };
+  // #endregion
+
+  // #region add todo and check
+  const handleAddTodo = (event: React.FormEvent) => {
+    event.preventDefault();
+    setTitleError(!title);
+    setUserIdError(!userId);
+
+    if (!title || !userId) {
+      return;
+    }
+
+    const createTodo: Todo = {
+      id: uniqueId(),
+      title,
+      completed: false,
+      userId,
+    };
+
+    addTodo(createTodo);
+
+    clearInputs();
+  };
+  // #endregion
+
+  const enrichedTodos = currentTodos.map(todo => ({
+    ...todo,
+    user: usersFromServer.find(user => user.id === todo.userId),
```

Comment:
Using `usersFromServer` directly here is not ideal. Consider passing the user data as props to this component to maintain a clear data flow and separation of concerns.